docs/architecture.md

```diff
@@ -239,3 +239,28 @@ NetworkPolicy implementation.
 
 As described earlier, Antrea Controller leverages the Kubernetes apiserver
 library to build the API and communication channel to Agents.
+
+### IPSec encryption
+
+Antrea supports encrypting GRE tunnel traffic with IPSec ESP. The IPSec
+implementation leverages [OVS IPSec](http://docs.openvswitch.org/en/latest/tutorials/ipsec)
+and leverages [strongSwan](https://www.strongswan.org) as the IKE daemon.
+
+To enable IPSec, an extra container -`antrea-ovs-ipsec` - must be added to the
+Antrea Agent DaemonSet, which runs the `ovs-monitor-ipsec` and strongSwan
+daemons. Antrea now supports only using pre-shared key (PSK) for IKE
+authentication, and the PSK string must be passed to Antrea Agent using an
+environment variable - `ANTREA_IPSEC_PSK`. The PSK string can be specified in
+the [Antrea IPSec deployment yaml](/build/yamls/antrea-ipsec.yml), which creates
```

Comment:
In practice, would people copy the PSK string to the YAML or would is it more common to create the secret separately through some other means (e.g. using `kubectl`)?

Reply:
I assume user will just use our yaml, as it is most convenient, and I do not feel any issue with that either.

Reply:
sounds good